lib/authn.js

```diff
@@ -128,7 +128,7 @@ async function _isAuthnRequired({config, admin}) {
 
 function _addAuthOptions({options, config}) {
   options.connectOptions.auth = {
-    user: config.username,
+    username: config.username,
```

Comment:
Reminder: We will be removing all these options in the next major release and just using `url`.

Reply:
I actually don't recall agreeing to that, I recall suggesting that we follow the mongo upgrade advice and either use object auth or url auth, but never a mix of both. Regardless this commit fixes the auth tests which saves us 30 minutes of tests that weren't timing out.

Reply:
Having just one way to do things is better. Unless there's some extra feature we get (which we will need to debate the usefulness) by having two ways, we should just have one way. The simplest way to do that when configuring is with a single `url` value.

Reply:
As long as we have a `connectOptions` someone can use an object to authorize. Anyways, let's discuss this more in the next PR which removes:

1. All defaults for `authSource`
2. `username` and `password` are removed from config
3. All of the auth related object construction is gone